_notices/rsn0042.md

```diff
@@ -0,0 +1,60 @@
+---
+layout: notice
+parent: RAPIDS Support Notices
+grand_parent: RAPIDS Notices
+nav_exclude: true
+notice_type: rsn
+# Update meta-data for notice
+notice_id: 42 # should match notice number
+notice_pin: true # set to true to pin to notice page
+
+title: "Moving older RAPIDS conda packages to a `legacy` label"
+notice_author: RAPIDS Ops
+notice_status: Proposal
+notice_status_color: blue
+# 'notice_status' and 'notice_status_color' combinations:
+#   "Proposal" - "blue"
+#   "Completed" - "green"
+#   "Review" - "purple"
+#   "In Progress" - "yellow"
+#   "Closed" - "red"
+notice_topic: Platform Support Change
+notice_rapids_version: "<=v23.02"
+notice_created: 2025-01-27
+# 'notice_updated' should match 'notice_created' until an update is made
+notice_updated: 2025-01-27
+---
+
+## Overview
+
+RAPIDS will move certain older conda packages in the `rapidsai` conda channel
+under a new label: `legacy`. These packages will still be available by adding
+`-c rapidsai/label/legacy` to installation and search commands.
+
+By moving these packages out of the main portion of the channel, newer RAPIDS
+conda environments will be solvable with `strict_channel_priority`, speeding up
+environment solve times.
+
+All RAPIDS conda install commands for RAPIDS versions `v23.04+` will continue to
+work without any user intervention.
+
+This change will occur concurrently with the release of RAPIDS `v25.04`.

```

Comment:
Just checking: We will be able to test strict channel priority in our CI before the 25.04 release, right? I want to be sure we minimize the risk of breaking recent releases (25.02 / 25.04) during this change.

Reply:
We'll be able to test `25.02` once all of those packages are released using the same method I tested the previous releases with.  Regarding 25.04, we can also release all of those packages, test, then move the selected older packages under the `legacy` label, or we can move them under the `label`, see if it works, and then undo the labeling if it breaks.  

If `25.02` works, I'd feel confident moving the packages straight to the `label` for 25.04, and that would let us accurately advertise strict channel priority for the new install instructions.

Reply:
That's helpful. What about enabling strict priority in our CI builds/tests, though? It sounds like you're discussing the user experience that we expect after the release is complete.

Reply:
We can't really enable them for CI until we move the older packages under `legacy`.  The `conda-metachannel` package is quite brittle at this point and I don't think we want that spun up as part of our CI

Reply:
Does that mean we will make the `legacy` label along with the 25.04 release, and then enable strict priority in our CI during development for 25.06?

Reply:
Yes

Reply:
You're asking a good question. I think the only confusing part here is timing.

We can't enable strict channel priority until we move "legacy" packages (needed for releases prior to 23.04) to the `legacy` label. There are packages like `datashader` that are implicated here.

We plan to move things to the `legacy` label _when 25.04 is released_. Then we can enable strict priority in 25.06.

However, there is a path where we enable strict priority _gradually in each repo's CI_, which I think we could start in 25.04.

Reply:
> Why is this? I'd think that in RAPIDS CI for 25.04, the pins like ==25.4.* we have for all RAPIDS packages would be sufficient to allow this.

channel priority doesn't consider versions -- if there is ANY version of a package in a higher priority channel, that is the only package that gets considered.  because we have (especially) a patched version of `datashader` (although there are other problem packages) in `rapidsai`, most of our solves won't work because it'll see `datashader` in `rapidsai` but the single version in `rapidsai` won't satisfy the current dependency requirements and the solve will fail.

Reply:
ahhhhhhh ok, thank you both for explaining. Ok then this makes sense to me as-is.

Agree with @bdice that it'd be really helpful to try to enable this repo-by-repo within the 25.04 cycle, to hopefully prevent accidental introduction of even *more* things that complicate the move to strict channel priority. And because it's a solution to the "did not install the package built in CI" problem: https://github.com/rapidsai/build-planning/issues/14#issuecomment-2659761340